test/AddTacticDoc.lean

```diff
@@ -0,0 +1,61 @@
+import Mathlib.Tactic.AddTacticDoc
+open Lean Elab
+
+/--My docstring for :one-/
+def one   := 1
+def two   := 2
+/--My docstring for :three-/
+def three := 3
+
+def tde : TacticDocEntry := {
+  name       := "something outlandish",
+  category   := DocCategory.hole_cmd,
+  decl_names := [`one],
+  tags       := ["not", "necessarily", "existent"]
+}
+
+def tde2 : TacticDocEntry := {
+  name       := "something outlandish",
+  category   := DocCategory.hole_cmd,
+  decl_names := [`one],
+  inherit_description_from := `three
+}
+
+-- 1. Test add_tactic_doc: use meta docstring
+/--meta!-/
+add_tactic_doc tde
+
+#eval show TermElabM _ from do
+  let tdes ← tactic.getTacticDocEntries
+  guard (tdes.length == 1)
+  if let some tde := tdes[0]? then
+    guard (tde.snd == "meta!")
+
+-- 2. Test add_tactic_doc: use the docstring of a single declaration from decl_names
+add_tactic_doc tde
+
+#eval show TermElabM _ from do
+  let tdes ← tactic.getTacticDocEntries
+  guard (tdes.length == 2)
+  if let some tde := tdes[0]? then
+    guard (tde.snd == "My docstring for :one")
+
+-- 3. Test add_tactic_doc: use inherit_description_from
+add_tactic_doc tde2
+
+#eval show TermElabM _ from do
+  let tdes ← tactic.getTacticDocEntries
+  guard (tdes.length == 3)
+  if let some tde := tdes[0]? then
+    guard (tde.snd == "My docstring for :three")
+
+-- 4. Test add_tactic_doc: create the tacticDocEntry on the fly
+add_tactic_doc {
+  name       := "never defined separately",
+  category   := DocCategory.hole_cmd,
+  decl_names := [`one]
+}
```

Comment:
Can it also support the following syntax:
```suggestion
add_tactic_doc where
  name       := "never defined separately"
  category   := DocCategory.hole_cmd
  decl_names := [`one]
```
You can implement this as a macro that expands to the version with the { }.

Reply:
It can't!  
Not super sure how to do it though (I'm porting another tactic at the moment, so I might only return to learning macros in a few weeks, wouldn't want this PR to go stale in the meantime).

Something along these lines?

```
macro metaDocString:docComment ? "add_tactic_doc " tdeStx:Lean.Parser.Command.whereStructInst : command => do
  `(add_tactic_doc $tdeStx)
```

Reply:
kind of, except that the macro you just wrote calls itself recursively. You would need to convert the where expression into a structure declaration with braces so that you can call the other version of the declaration.